src/wazuh_db/wdb_agent.c

```diff
@@ -133,59 +172,134 @@ int wdb_remove_agent(int id) {
     char wdbquery[OS_BUFFER_SIZE] = "";
     char wdboutput[OS_BUFFER_SIZE] = "";
     int wdb_sock = -1;
-    char * name = "";
+    char * name = NULL;
 
     name = wdb_agent_name(id);
-    snprintf(wdbquery, OS_BUFFER_SIZE, global_db_queries[SQL_DELETE_AGENT], id);
-    result = wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput)) == 0;
-    wdb_delete_agent_belongs(id);
+    sqlite3_snprintf(sizeof(wdbquery), wdbquery, global_db_queries[SQL_DELETE_AGENT], id);
+    result = wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput));
 
-    result = result && name ? wdb_remove_agent_db(id, name) : -1;
+    switch (result){
+        case OS_SUCCESS:
+            wdb_delete_agent_belongs(id);
+            result = name ? wdb_remove_agent_db(id, name) : OS_INVALID;
+            if(result == OS_INVALID){
+                mdebug1("Unable to remove agent DB: %d - %s ", id, name);
+            }
+            break;
+        case OS_INVALID:
+            mdebug1("GLobal DB Error in the response from socket");
+            mdebug2("Global DB SQL query: %s", wdbquery);
+            break;
+        default:
+            mdebug1("GLobal DB Cannot execute SQL query; err database %s/%s.db", WDB2_DIR, WDB2_GLOB_NAME);
+            mdebug2("Global DB SQL query: %s", wdbquery);
+            break;
```

Comment:
The `break` is not necessary in the `default` case.

Reply:
Done